kore/src/Kore/Simplify/Simplify.hs

```diff
@@ -315,8 +315,12 @@ simplifyPatternScatter ::
     SideCondition RewritingVariableName ->
     Pattern RewritingVariableName ->
     simplifier (Pattern RewritingVariableName)
-simplifyPatternScatter sideCondition patt =
-    simplifyPattern sideCondition patt >>= Logic.scatter
+simplifyPatternScatter sideCondition patt = do
+    simplifierX <- askSimplifierXSwitch
+    Logic.scatter
+        =<< case simplifierX of
+            EnabledSimplifierX -> simplifyPatternId patt
```

Comment:
I'm not sure if we care about clarity here since this is temporary anyway. `simplifyPatternId` will never split, so it always returns an `OrPattern` with one alternative, right? In turn, that means `Logic.scatter` is a no-op. I think
```hs
simplifierX <- askSimplifierXSwitch
case simplifierX of
    EnabledSimplifierX -> pure patt
    DisabledSimplifierX -> 
        simplifyPattern sideCondition patt >>= Logic.scatter
```
is clearer.

Reply:
I think we should keep `simplifyPatternId` in the source code for grep, but I also think we should avoid `OrPattern.toPattern` because of https://github.com/kframework/kore/issues/2610